### PR TITLE
Idea related to a mentee mentor recommendation system

### LINKED ISSUE #785: Come up with a possible solution for building a Mentor Mentee Recommendation and write a technical article on it.
**Description**
As a contributor I want you to come up with a possible solution of Mentee Recommendation to a Mentor. Please be very clear in your approach on what algorithm you are using and show it's proper application on how this task can be done here. 

Design the solution submit as a markdown file under the “docs” folder.

**Acceptance Criteria**

- [ ] Update [Required]

- [ ] Submit as a markdown file under the “docs” folder.

**Definition of Done**

- [ ] All of the required items are completed.

- [ ] Approval by 2 mentors.

**Estimation**
4+ hours

**Note**
This issue will be assigned from September 1, 2020.

### PR DESCRIPTION
**Pull request by Anmollenka**
### Description

A solution for recommending mentees to a mentor.

Fixes #785 

### Type of Change:

- Code
- Outreach

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
I have previewed the code in the editor.
Screenshot (1038)

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Anmollenka** [comment]
@yashasvimisra2798 Can you please review my pull request.Thank You.

**Review by yashasvimisra2798 — COMMENTED** [review]
Is it only the programming languages or the fields as well?
eg: app development, computer vision etc

**Comment by Anmollenka** [comment]
@yashasvimisra2798 I have taken only the programming languages as I was not sure of what are the different fields to be taken i.e what career developement topics in which mentoring are to be given.

**Comment by Anmollenka** [comment]
Should I make any changes??

**Review by yashasvimisra2798 — DISMISSED** [review]
It's ok with the programming languages. But having the skills would have been better.

**Comment by Anmollenka** [comment]
Shall I make chan

> It's ok with the programming languages. But having the skills would have been better.

Shall I make the changes or is it okk??

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@Anmollenka please squash your commits into one. Also follow the commit guidelines. Feel free to change your pull request title too

**Comment by Anmollenka** [comment]
@devkapilbansal I will make the required changes.

**Comment by Anmollenka** [comment]
@devkapilbansal I have squashed the commits.

**Comment by Anmollenka** [comment]
@devkapilbansal I tried squashing the commits into one but I don't know from where 2 commits are showing.Can you please help me with this.

**Comment by devkapilbansal** [comment]
@Anmollenka follow these steps
```
At first fetch the new changes from develop branch
git remote add upstream https://github.com/anitab-org/mentorship-backend.git
git fetch upstream develop
git reset --soft upstream/develop
git commit
```
Alternatively you can try removing the two commits you made today and then squashing and rebasing
``` 
git reset --hard 6a4ba12
git reset --soft HEAD^4
git commit
git rebase upstream/develop(make sure you add upstream in remote as asked above)
```

**Comment by Anmollenka** [comment]
@devkapilbansal I have made the required changes.Please review it.Thank You.

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@Anmollenka it would be better if you upload your images to Github and use links instead. These binary files put unnecessary weights to the repository. 
Here is the guide on how to do that:
https://gist.github.com/vinkla/dca76249ba6b73c5dd66a4e986df4c8d

Also, feel free to edit the PR title to make it more descriptive

**Comment by Anmollenka** [comment]
@devkapilbansal I had used the images  to explain my process through code in the markdown file,so I decided to upload the images in this repository.Do you want that I should upload  the images in some github repository of mine and then use a link.

**Comment by devkapilbansal** [comment]
> @devkapilbansal The images to explain my process through code in the markdown file.So I decided to upload the images in this repository.Do you want that I should upload the images in some github repository of mine and then use a link.

Yes, it will not affect your markdown file too and reduce weight on repo at the same time

**Comment by Anmollenka** [comment]
@devkapilbansal I have made the required changes.Please review it.Thank You.

File: docs/Recommendation.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Mentorship System is an application that allows women in tech to mentor each other, on career development topics,through 1:1 relations for a certain period. As we know that the mentorship can be provided on any career development topics but for the ease of explaining my concepts I have used topics which are basically programming languages that are being used in developement applications for mentorship systems.
+<br>
+There can be two methods to form the mentee-mentor DataFrame.The original dataset can be taken from the mentorship system database.
+
+### FIRST METHOD
+1. We will create a mentee Dataframe which basically contains the percentage of knowledge a particular mentee has on the given programming languages .If the mentee has not mentioned any knowledge level for any programming language then that has to be taken 0.But necessarily every mentee should have filled some knowledge percentage on the  given skills.Below is the sample Dataframe:-
+<br>
+![Screenshot (1032)](https://user-images.githubusercontent.com/48198809/97035787-79e74c80-1584-11eb-86bd-6970f54bca9a.png)
+<br>
```
Comment (devkapilbansal) [inline docs/Recommendation.md]: @Anmollenka we can achieve same results by leaving the blank line too. No need to add <br>

File: docs/Recommendation.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+![Screenshot (1031)](https://user-images.githubusercontent.com/48198809/97035706-5ae8ba80-1584-11eb-889b-3a76500826c9.png)
+<br>
+I am using collaborative filtering method for recommender system . It is basically a user-user collaborative filtering which gives more accurate results . Below is the code of the recommender system.
+<br>
+![Screenshot (1036)](https://user-images.githubusercontent.com/48198809/97035893-a13e1980-1584-11eb-96c8-912dcfe84c8e.png)
+<br>
+As we can see here that the mentees recommended here to the mentors in  they are weak particularly in areas where the mentors have good knowledge and have average or good knowledge at areas where the mentors have less knowledge in descending order so that they wouldn't face any issue overally.
+
+
```
Comment (devkapilbansal) [inline docs/Recommendation.md]: Why two blank lines here?One is enough here

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@Anmollenka some of your images are not visible in the markdown file. You can verify it manually by going to [its github link](https://github.com/Anmollenka/mentorship-backend/blob/new/docs/Recommendation.md).

**Comment by Anmollenka** [comment]
@devkapilbansal I am able to preview it on my side.
Screenshot (1098)
I am not able to understand where I am going wrong.Thank You.

**Comment by Anmollenka** [comment]
> @devkapilbansal I am able to preview it on my side.
> Screenshot (1098)
> I am not able to understand where I am going wrong.

**Comment by devkapilbansal** [comment]
@Anmollenka that `br` tags are making such differences. Use their alternatives

**Comment by Anmollenka** [comment]
OK.I will make the changes

**Comment by Anmollenka** [comment]
@devkapilbansal I have made the required changes.Please review it.

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@Anmollenka use any online tool to proofread the document and correct grammar/spelling error.
Grammarly can be good for that

**Comment by Anmollenka** [comment]
> @Anmollenka use any online tool to proofread the document and correct grammar/spelling error.
> Grammarly can be good for that

@devkapilbansal I will keep this in mind.

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@Anmollenka can you please squash your commit in one

**Review by devkapilbansal — APPROVED** [review]

**Comment by vj-codes** [comment]
@Anmollenka thank you for your contribution 
We will surely visit this PR when implementing the feature:)

**Pull request closed without merging**